setup.py

```diff
@@ -98,23 +111,77 @@ def get_geos_paths():
 
 
 # Add numpy include dirs without importing numpy on module level.
-# See https://stackoverflow.com/questions/19919905/
-# how-to-bootstrap-numpy-installation-in-setup-py/21621689#21621689
+# derived from scikit-hep:
+# https://github.com/scikit-hep/root_numpy/pull/292
 class build_ext(_build_ext):
     def finalize_options(self):
         _build_ext.finalize_options(self)
         # Prevent numpy from thinking it is still in its setup process:
-        __builtins__.__NUMPY_SETUP__ = False
+        try:
+            del builtins.__NUMPY_SETUP__
+        except AttributeError:
+            pass
+
         import numpy
 
         self.include_dirs.append(numpy.get_include())
 
 
-module_lib = Extension(
-    "pygeos.lib",
-    sources=["src/lib.c", "src/geos.c", "src/pygeom.c", "src/ufuncs.c", "src/coords.c", "src/strtree.c"],
-    **get_geos_paths()
-)
+ext_modules = []
+
+if "clean" not in sys.argv:
+    ext_options = get_geos_paths()
+
+    ext_modules = [
+        Extension(
+            "pygeos.lib.core",
+            sources=[
+                "src/core.c",
+                "src/c_api.c",
+                "src/coords.c",
+                "src/geos.c",
+                "src/pygeom.c",
+                "src/strtree.c",
+                "src/ufuncs.c",
+            ],
+            **ext_options,
+        )
+    ]
+
+    if os.path.exists("MANIFEST.in"):
+        # likely building from source; Cython is required
+
+        if not cythonize:
+            sys.exit("ERROR: Cython is required to build pygeos from source.")
+
+        # numpy libs must be included for Cython build
+        import numpy
+
+        ext_options["include_dirs"].append(numpy.get_include())
```

Comment:
I think this is arranged already in https://github.com/pygeos/pygeos/pull/197/files#diff-2eeaed663bd0d25b7e608891384b7298R127